src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Semantics;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+
+namespace Bicep.Core.Analyzers.Linter
+{
+    internal class LinterAnalyzer : IBicepAnalyzer
+    {
+        public static string AnalyzerName => "Bicep Internal Linter";
+        public static readonly bool IsCliInvoked;
+
+        private static readonly IEnumerable<Type> RuleTypes;
+        private static readonly ImmutableArray<IBicepAnalyzerRule> RuleSet;
+
+        static LinterAnalyzer()
+        {
+            var entryAssembly = Assembly.GetEntryAssembly();
+            IsCliInvoked = entryAssembly.Location.EndsWith("bicep.dll", true, CultureInfo.CurrentCulture);
+            
+            RuleTypes = Assembly.GetExecutingAssembly()
+                .GetTypes()
+                .Where(t => t.IsSubclassOf(typeof(LinterRule)));
```

Comment:
Here you check for `typeof(LinterRule)`, but in `CreateLinterRules()`, you check for `typeof(IBicepAnalyzerRule)` - worth making this consistent?

Reply:
Want to skip the LinterRule base class. I can swap over to finding the interface, so that new rules don't have to use the Base class.